docs/design/proposals/direct-blk-device-assignment.md

```diff
@@ -0,0 +1,264 @@
+different cluster admins can choose how to specify this depending on their use case. For example, the behavior can be specified 
+by a storage class parameter or CSI plugin lookup pod runtime class. 
+
+
+## Interface between components
+
+Here we describe the changes we need to make in different components in order for the volume information to be propagated 
+from CSI to the Kata container runtime. 
+
+### Custom CSI node driver changes
+In the `skip-hostmount` mode, CSI node driver defers the filesystem mount operation to the container runtime. 
+The implementation of a few node driver APIs needs to be updated.
+
+* **NodePublishVolume** -- It propagates the mount information to the container runtime for it to carry out filesystem mount operation.
+* **NodeGetVolumeStats** -- Since the container runtime mounts the filesystem, it needs to forward the request to the container runtime.
+* **NodeExpandVolume** -- The CSI cannot implement this API alone. It needs to propagate the new size to the container runtime so that the filesystem metadata can be refreshed accordingly.
+* **NodeStageVolume/NodeUnStageVolume** -- Invoke a kata-runtime handler to clean up the stale mount information.
+Typical block device based CSI plugins (like EBS, Azure, GCE PD, Ceph, etc) 
+mounts the PV to a global mount path in the host during NodeStageVolume, which can be skipped in this case.
+
+### Kata-runtime changes
+The host agent (CSI node driver, as an example) will call into kata-runtime. Kata-runtime behaves as the API for 
+sandbox-level management interactions. In order for the CSI node driver to communicate with the kata-runtime, 
+we propose augmenting kata-runtime to include a filesystem, `fs`, command as follows:
+
+```
+kata-runtime fs —help
+NAME:
+kata-runtime fs [subcommand] [options]— interact with mounted filesystem volume
+
+SUBCOMMANDS:
+ - add-blk-mount-info <mountinfo> -- add the mount info of a volume to the Kata runtime.
+ - reset-blk-mount-info <host-blk-device-name> –- clean up the mount info directory under kata runtime
+ - get-stats <host-blk-device-name> — show status and usage details for the mounted filesystem backed by the given block device, <host-blk-device-name>.
+ - resize <host-blk-device-name> <size> — update the mounted volume to <size> for the mounted filesystem backed by the given block device, <host-blk-device-name>.
+
+```
+The mountinfo object is defined as follows:
+```Golang
+type MountInfo struct {
+    // Device: source device path for this volume
+    Device string json:"device,omitempty"
+    // FsType: filesystem that needs to be used to mount the storage inside the VM
+    FsType string json:"fsType,omitempty"
+    // metadata field for easy extension
+    Metadata map[string]string json:"metadata,omitempty"
+    // Options: additional options that might be needed to mount the storage filesystem
+    Options []string json:"options,omitempty"
```

Comment:
Given this information gets persisted on disk as part of the implementation, how about a specific note here indicating `options` should not include any secrets (that may have been passed through CSI). For example, SMB mount password.